src/OrbitClientModel/CaptureData.cpp

```diff
@@ -65,14 +65,27 @@ void CaptureData::UpdateFunctionStats(const FunctionInfo& function, uint64_t ela
   }
 }
 
-const FunctionInfo* CaptureData::GetSelectedFunction(uint64_t function_address) const {
-  auto selected_functions_it = selected_functions_.find(function_address);
-  if (selected_functions_it == selected_functions_.end()) {
+const FunctionInfo* CaptureData::GetInstrumentedFunctionById(uint64_t function_id) const {
+  auto selected_functions_it = instrumented_functions_.find(function_id);
+  if (selected_functions_it == instrumented_functions_.end()) {
     return nullptr;
   }
   return &selected_functions_it->second;
 }
 
+std::optional<uint64_t> CaptureData::GetInstrumentedFunctionId(
+    const orbit_client_protos::FunctionInfo& function) const {
+  for (const auto& it : instrumented_functions_) {
```

Comment:
Can this become a performance bottleneck? If so, we could arrange the target functions sorted by address and do a binary search.

Reply:
At the moment this function is only used when we select a new function. never during capture or when drawing the track. I would rather consider not using it than optimizing this implementation.

Reply:
Then maybe rename it? e.g. `ComputeInstrumentedFunctionId`/`FindInstrumentedFunctionId` and maybe add a comment?

Reply:
Done.